lib/concepts/lesson/operations/update.rb

```diff
@@ -8,13 +8,32 @@ def call(lesson:, lesson_params:)
         response[:lesson] = lesson
         response[:lesson].assign_attributes(lesson_params)
         response[:lesson].save!
+        if lesson_params[:name].present?
+          rename_lesson_project(lesson: response[:lesson], name: lesson_params[:name])
+          rename_lesson_remixes(lesson: response[:lesson], name: lesson_params[:name])
+        end
         response
       rescue StandardError => e
         Sentry.capture_exception(e)
         errors = response[:lesson].errors.full_messages.join(',')
         response[:error] = "Error updating lesson: #{errors}"
         response
       end
+
+      def rename_lesson_project(lesson:, name:)
+        return unless lesson.project
+
+        lesson.project.update!(name:)
+      end
+
+      def rename_lesson_remixes(lesson:, name:)
+        return unless lesson.project
+
+        lesson_remixes = Project.where(remixed_from_id: lesson.project.id)
```

Comment:
would it be worth considering a transaction here? just to ensure we don't end up with some renamed and some not which might be messy to unpick

Reply:
ooh yes I hadn't thought of that 👀

Reply:
Okay, that should be done 👍